packages/gatsby/src/cache-dir/production-app.js

```diff
@@ -77,11 +77,13 @@ window.___navigateTo = navigateTo
 const history = createHistory()
 
 function attachToHistory(history) {
-  window.___history = history
+  if(!window.___history) {
+    window.___history = history
 
-  history.listen((location, action) => {
-    apiRunner(`onRouteUpdate`, { location, action })
-  })
+    history.listen((location, action) => {
+      apiRunner(`onRouteUpdate`, { location, action })
+    })
+  }
```

Comment:
yeah, root.js is for the development version of Gatsby. It should be fixed too but since you're not generally tracking yourself in development it's not as urgent.